authentication_sasl.go

```diff
@@ -58,3 +59,27 @@ func (src *AuthenticationSASL) Encode(dst []byte) []byte {
 
 	return dst
 }
+
+// MarshalJSON implements encoding/json.Marshaler.
+func (src AuthenticationSASL) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		Type           string
+		AuthMechanisms []string
```

Comment:
Maybe embedding is better?

```suggestion
		Type           string
		AuthenticationSASL
```

Reply:
I don't think this will work since it will lead to infinite recursion when marshaling itself.